oop-programs_with_interesting_classes-easy-q4 (#712)
canvas-gamification/canvas-gamification-junit-tests: src/oop/programs_with_interesting_classes/easy/q4/QuestionText.md

# oop-programs_with_interesting_classes-easy-q4

- Parsons Question

## Question Text

> Life is a beautiful thing, and as time passes, things grow. Write a Java class about plants called `Tree`.

Reviewer: ```suggestion
Life is a beautiful thing, and as time passes, things grow. Write a Java class about plants called `Tree` to model a plant with the requirements below:
```

### Tree Class

The `Tree` class has the following requirements:

- Attributes
    - Note that none of the attributes in this class should be visible from another class
    - A `String` attribute called **type**
    - A `double` attribute called **size**

- Constructors:
    - A constructor visible from another class that accepts a `String` and a `double` as parameters, and uses the values
      to initialize the **type** and **size** attributes
    - Note that this constructor should be visible from all other classes.

- Methods:
    - A accessor and mutator methods for the **type** attribute called `getType` and `setType`.

>     - A accessor and mutator methods for the **size** attribute called `getSize` and `setSize`.

Reviewer: I think we should change this wording so we are consistent with the rest of the questions.

Reviewer: Also please include the return type for the `set` methods

# oop-programs_with_interesting_classes-easy-q4

- Parsons Question

## Question Text

> Life is a beautiful thing, and as time passes, things grow. Write a Java class about plants called `Tree` to model a plant with the requirements below:

Reviewer 2: ```suggestion
Life is everywhere on the planet, and it is constantly changing. Write a Java class about plants called `Tree` to model a plant with the requirements below:
```

### Tree Class

The `Tree` class has the following requirements:

- Attributes
    - Note that none of the attributes in this class should be visible from another class
    - A `String` attribute called **type**
    - A `double` attribute called **size**

- Constructors:
    - A constructor visible from another class that accepts a `String` and a `double` as parameters, and uses the values
      to initialize the **type** and **size** attributes
    - Note that this constructor should be visible from all other classes.

> - Methods:

Reviewer 2: make a note of their visibility

>     - A `gettingOlder` method which addes 4 to the **size** attribute.

Reviewer 2: ```suggestion
    - A `gettingOlder` method which adds 4 to the **size** attribute.
```
